packages/app/src/specs/banners/TrackedBanner.vue

```diff
@@ -67,7 +94,7 @@ const setStateMutation = useMutation(TrackedBanner_SetProjectStateDocument)
 const reportSeenMutation = useMutation(TrackedBanner_RecordBannerSeenDocument)
 const bannerInstanceId = ref(nanoid())
 
-watchEffect(() => {
+watch(() => props.eventData?.cohort, () => {
```

Comment:
This feels wrong. What about a situation where the banner is initially not visible (`modelValue === false`) then later made visible - this watcher won't fire again for that change, and we'll never record the event.

Reply:
Good catch there.  I had to change this from the watchEffect to the watch because I did not want it to fire before the cohort was set.  I will test out the scenario of when that modalValue changes and see if it records the event.  The logic is definitely not as clear cut as I would like, but I think it will still work.  Will the `props.modalValue` ever be false here?  I believe the `v-else-if` on the `LoginBanner` in the `SpecsListBanners` will prevent this code from running unless the `props.modalValue` is already true, right?

Reply:
Given how things are currently structured, yes, `v-model` will always *happen* to be true when initially rendered, but there's no guarantee we won't add a new banner next month someplace else that doesn't rely on a `v-if` and uses `v-model` dynamically. There's no way at this layer for the component to tell the difference between "I have no cohort" vs "I haven't yet received a value for cohort" - that would need to be communicated by refactoring this (another prop to communicate `hasCohortResolved`, or a composable that uses `undefined` vs `null` to communicate "none" vs "not yet".

In any event, this watcher now has a couple gaps that are relying on race conditions. What happens if cohort resolves before the banner displays? The watcher will never fire because it doesn't have `immediate: true` - that's part of the reason it was a `watchEffect` before

Reply:
So, the current issue which breaks the recording of the event is that the `hasBannerBeenShown` is first false but then set to true before this watchEffect gets triggered.  Rethinking how to order these operations to prevent this race condition.